### DISCUSSION TITLE
Cant login to google accounts

### DISCUSSION THREAD

**Discussion by jarno9981** [cefsharp/CefSharp#3506] [General]
Cefsharp latest version
.net framework 4.7.2 
Windows 10

Can't login to google accounts
Google say its usafe please fix 
So we can login to google accounts again

**Comment by amaitland** [comment]
> Starting in June 2019 Google will block logins from CEF based browsers to Google Services, this includes Gmail, Drive, Docs, see https://security.googleblog.com/2019/04/better-protection-against-man-in-middle.html?m=1 for further discussion see https://groups.google.com/a/chromium.org/d/msg/embedder-dev/STyM5ZNTHMM/POj1v_cqBgAJ

As detailed in the [Release Notes]( https://github.com/cefsharp/CefSharp/releases/tag/v89.0.170)

See also
https://magpcss.org/ceforum/viewtopic.php?f=10&t=16764
https://magpcss.org/ceforum/viewtopic.php?f=10&t=16717

> Google say its usafe please fix

Some users have chosen to use the Browser for illegal/questionable activity. Man in the middle attacks aren't a security flaw in the browser,  it's how people have chosen to  use the browser. This is not something I can control/fix.

**Reply by jarno9981** [reply]
Maybe add google oauth to maybe that works (suggastion)

**Reply by amaitland** [reply]
If anyone successfully implements logging into Google with OAUTH then please provide a sample as reference to others. 

I've seen reference to users attempting such, nobody has provided an example. 

https://magpcss.org/ceforum/viewtopic.php?f=10&t=18006

**Reply by jarno9981** [reply]
https://github.com/googlesamples/oauth-apps-for-windows

^^google o auth sample 

https://www.c-sharpcorner.com/news/google-releases-guide-on-setting-up-oauth-20-flows

^^page with help about google oauth

**Reply by jarno9981** [reply]
https://developers.google.com/identity/protocols/oauth2/native-app

Official docs for browser based desktop apps oauth

**Reply by amaitland** [reply]
Personally I have no plans to look into this, a pull request with an example would be welcome.

**Reply by amaitland** [reply]
As part of your investigation you have any CefSharp specific questions then I'll answer if possible.
